tests/cpp/engine/threaded_engine_test.cc

```diff
@@ -121,6 +121,19 @@ double EvaluateWorloads(const std::vector<Workload>& workloads,
   return dmlc::GetTime() - t;
 }
 
+TEST(Engine, start_stop) {
+  const int num_engine = 3;
+  std::vector<mxnet::Engine*> engine(num_engine);
+  engine[0] = mxnet::engine::CreateNaiveEngine();
+  engine[1] = mxnet::engine::CreateThreadedEnginePooled();
+  engine[2] = mxnet::engine::CreateThreadedEnginePerDevice();
+
+  for (int i = 0; i < num_engine; ++i) {
+    engine[i]->Stop();
+    engine[i]->Start();
```

Comment:
Could you verify whether the actions have actually been successful?

Reply:
Without adding new API, I cannot think of a way to verify except sanity test.